libp2p/RLPxHandshake.cpp

```diff
@@ -450,7 +452,35 @@ void RLPXHandshake::transition(boost::system::error_code _ech)
                                         << "Invalid packet type. Expected: "
                                         << p2pPacketTypeToString(HelloPacket)
                                         << ", received: " << p2pPacketTypeToString(packetType);
-                                    m_failureReason = ProtocolError;
+                                    m_failureReason = HandshakeFailureReason::ProtocolError;
+                                    if (packetType == DisconnectPacket)
```

Comment:
Better add a comment to explain why we do this (Disconnect is allowed at any moment, so it's not a protocol violation and we should try to reconnect later)